### ISSUE TITLE
why no sample web application project to demonstrate about 3d security v2?

### ISSUE THREAD

**Issue by Jey-Vclub** [globalpayments/dotnet-sdk#32]
I have seen all your examples project. all are build using your very old api.

coming to the point.

1. i read your developer portal about 3d security v2 implementation using .net but, there lot of continuity is missing
2. most of the terms are not explained
3. on your sample code section, you just posted portion of code only. 100% no one can understand this type.

i recommend to you is, please post step by step practical guide project in github because, your developer portal is not at all enough to understand the flow. 

kindly reply me since, i called you many time but, no response; email also no proper response.

thank you
jey

**Comment by seanmacdomhnall-gp** [comment]
Hi @Jey-Vclub 

1. An end to end example is available in the Documentation section of the portal:
https://developer.globalpay.com/ecommerce/3d-secure-version2#api

2. All of the terms are outlined in the API Explorer of the portal:
https://developer.globalpay.com/api/3d-secure-two

3. If you scroll to the bottom of this article: https://developer.globalpay.com/ecommerce/3d-secure-version2#api, the complete code samples are available

Thanks for your feedback, we'll look at posting a project on GitHub.

**Comment by Jey-Vclub** [comment]
Hello there,

I have already read all the mentioned pages more than 4 times but, continuity is missing. So, please provide a project with proper example.

I did download SDK, there I could see examples\webforms\end-to-end\ but, there they used very old sdk without 3d security v2.

So, all we need is step by step source code to achieve 3d security v2. Kindly share,

Thanks & Regards

Jey
Technical Architect & Business Analyst

**Comment by jonathancounihan** [comment]
I have to agree with the OP, @Jey-Vclub the documentation and samples are lacking.

From this page: https://developer.globalpay.com/ecommerce/3d-secure-version2#api

> The first thing we need to do before testing or implementing 3D Secure 2 is setup and configure endpoints in your application/website that will receive data and event notifications from the Issuer Access Control Server (ACS). They need to be capable of accepting a HTTP POST with Base64 encoded values and are used in two key moments during authentication:

So why can't you show this controller method? What you expect and how it should be structured? 

I have also read these pages a few times and got no closer to getting 3DS V2 working.

**Comment by seanmacdomhnall-gp** [comment]
Hi @jonathancounihan the two notification endpoints are documented on that page:

- Sample Method URL Notification Endpoint (Server-Side)
- Sample Challenge Notification Endpoint (Server-Side)

Do these cover what you are looking for?

Thanks for your feedback, will pass it on to the relevant team.

**Comment by schofieldjraileasy** [comment]
The .NET "Complete Obtain Authentication Data & Authorization Example" on https://developer.globalpay.com/ecommerce/3d-secure-version2/browser-authentication#api appears to be incorrect and is an exact copy of the "Complete Initiate Authentication Example".

**Comment by anurit** [comment]
I also agree, documentation is poor and has bugs. 
https://developer.globalpay.com/ecommerce/3d-secure-version2/browser-authentication#api
PHP version has:
$threeDSecureData = Secure3dService::checkEnrollment($card)->execute(Secure3dVersion::TWO);
should be:
$threeDSecureData = Secure3dService::checkEnrollment($card)->execute('default', Secure3dVersion::TWO);
Still no one fixed it. It is   hard to integrate required 3D security 2 if documentation has bugs and is incomplete.

**Comment by badrshs** [comment]
the worst documentation ever :) . thanks realex for wasting ours time.

**Comment by Externaluse** [comment]
This issue has only been open for two years? I think we'll move to Stripe now, it's completely impossible to work with the API since Realex sold it. The stored card grandfathering doesn't work, the 2nd level support can't figure out why some transactions work and others don't, why the test system sometimes works fine in all flows when the live system blows up, and so on and so forth.
I'm throwing in the towel.

**Comment by badrshs** [comment]
that always the case :D, tons of bugs,  lots of missing payments, we lost 5 months of work then at the end they said it's not supported on production yet. Stripe always  are the best
